src/components/calcite-input-time-picker/calcite-input-time-picker.tsx

```diff
@@ -142,8 +142,8 @@ export class CalciteInputTimePicker {
     event.stopPropagation();
     if (event.detail) {
       const { hour, minute, second } = event.detail as Time;
-      if (hour !== "--" && minute !== "--") {
-        if (this.step !== 60 && second !== "--") {
+      if (hour !== null && minute !== null) {
```

Comment:
I think instead of checking explicitly for null, you should check to see if the type is a number. Then you can treat undefined/null/"", false as the same thing and have the defaults be undefined.

Reply:
I'll have to `parseInt` the value before checking because the values are stored as strings because of the zero padding, but I think this should be fine as a compromise.

Reply:
Weeeeird
![image](https://user-images.githubusercontent.com/821864/111709873-b7839500-8805-11eb-9ac6-01ef2d94d5be.png)

Reply:
I'll have to rely on my trusty `stringContainsOnlyNumbers` util helper.

Reply:
If you use parseInt, you'll need to use `isNaN()` to check it I think.

Reply:
Resolved by: https://github.com/Esri/calcite-components/pull/1736/commits/92a343e3451de3b8fedc72e3f332397d21b34b9a

Reply:
Resolved by: https://github.com/Esri/calcite-components/pull/1736/commits/92a343e3451de3b8fedc72e3f332397d21b34b9a